engine/src/main/java/org/pentaho/di/trans/steps/transexecutor/TransExecutor.java

```diff
@@ -374,6 +374,34 @@ void passParametersToTrans( List<String> incomingFieldValues ) throws KettleExce
         .activateParams( trans, trans, this, trans.listParameters(), parameters.getVariable(), inputFieldValues, meta.getParameters().isInheritingAllVariables() );
   }
 
+  /**
+   * Search through the provided list for a specified value while ignoring case
+   * @param list list of Strings to be searched
+   * @param value String to search for
+   * @return true if the value is found, false otherwise
+   */
+  private boolean listContainsIgnoreCase( final List<String> list, final String value ) {
+    return list.stream().anyMatch( s -> s.equalsIgnoreCase( value ) );
+  }
+
+  /**
+   * Get the index of the specified value in the list, ignoring case.  Should only be called
+   * after verifying that the value exists first using listContainsIgnoreCase!
+   * @param list list of Strings to be searched
+   * @param value String to search for
+   * @return the index of the value in the list if found, -1 if not found
+   */
+  private int getIndexIgnoreCase( final List<String> list, final String value ) {
+    for ( int i = 0; i < list.size(); i++ ) {
+      if ( list.get( i ).equalsIgnoreCase( value ) ) {
+        return i;
+      }
+    }
+    // THIS SHOULD NEVER HAPPEN BECAUSE WE SHOULD NEVER CALL THIS METHOD WITHOUT FIRST CHECKING FOR THE EXISTENCE OF
```

Comment:
Don't think this comment is necessary, ArrayList.indexOf() also returns -1 if not in the list

Reply:
I just want to be certain that the intent of the code is clear when someone else has to maintain it next.  If we wind up in the situation where this method has to return -1 (i.e. the current assumptions are broken), the current exception handling won't make it very clear what the actual problem is